jsonnetsecure/jsonnet_pool.go

```diff
@@ -129,27 +138,40 @@ func newWorker(ctx context.Context) (_ worker, err error) {
 	if err != nil {
 		return worker{}, errors.Wrap(err, "newWorker: failed to create stderr pipe")
 	}
+
+	stdoutReader := io.LimitReader(stdout, int64(jsonnetOutputLimit))
+	stderrReader := io.LimitReader(stderr, int64(jsonnetErrLimit))
+
 	if err := cmd.Start(); err != nil {
 		return worker{}, errors.Wrap(err, "newWorker: failed to start process")
 	}
 
 	span.SetAttributes(semconv.ProcessPID(cmd.Process.Pid))
 
-	scan := func(c chan<- string, r io.Reader) {
+	scan := func(c chan<- string, r io.Reader, limit int) {
 		defer close(c)
+		// NOTE: `bufio.Scanner` has its own internal limit of 64 KiB.
 		scanner := bufio.NewScanner(r)
+
 		scanner.Split(splitNull)
 		for scanner.Scan() {
-			c <- scanner.Text()
+			s := scanner.Text()
+			if len(s) >= limit {
+				c <- "ERROR: reached limits: " + strconv.FormatInt(int64(len(s)), 10)
+			} else {
+				c <- s
+			}
 		}
 		if err := scanner.Err(); err != nil {
 			c <- "ERROR: scan: " + err.Error()
 		}
 	}
-	out := make(chan string)
-	go scan(out, stdout)
-	errs := make(chan string)
-	go scan(errs, stderr)
+	out := make(chan string, chanSize)
```

Comment:
It would make sense to make `out` and `errs` buffered with size 1, to prevent theoretical goroutine leaks.